### PR TITLE
Remove Dry::Types::Struct and Dry::Types::Value in favor of Dry::Struct

### PR DESCRIPTION
**Pull request by backus**

### CODE REVIEW

**Comment by backus** [comment]
@solnic could you review this?

**Comment by solnic** [comment]
I think we should remove anything related to dry-struct and add struct-specific error handling to dry-struct itself. WDYT?

**Comment by backus** [comment]
Yeah I agree @solnic. I'll followup with that

**Comment by backus** [comment]
@solnic squeezed out the remaining Struct references

File: benchmarks/basic.rb [hunk 1]
```diff
@@ -41,7 +41,7 @@ def attributes=(attributes = {})
   end
 end
 
-class DryTypesUser < Dry::Types::Struct
+class DryTypesUser < Dry::Struct
   attributes(name: 'string', age: 'form.int')
 end
```
Comment (solnic) [inline benchmarks/basic.rb]: benchmarks could be moved to dry-struct too :)

**Comment by backus** [comment]
@solnic benchmarks removed

File: Gemfile [hunk 2]
```diff
@@ -22,3 +22,4 @@ group :benchmarks do
 end
 
 gem "dry-logic", git: 'https://github.com/dry-rb/dry-logic.git', branch: "master"
+gem 'dry-struct', git: 'https://github.com/dry-rb/dry-struct.git', branch: 'master'
```
Comment (solnic) [inline Gemfile]: This shouldn't be needed anymore.

File: lib/dry/types.rb [hunk 3]
```diff
@@ -12,8 +12,8 @@
 require 'dry/types/container'
 require 'dry/types/definition'
 require 'dry/types/constructor'
-require 'dry/types/struct'
-require 'dry/types/value'
+
+require 'dry/struct'
```
Comment (solnic) [inline lib/dry/types.rb]: No need to require it

File: spec/dry/types/array_spec.rb [hunk 4]
```diff
@@ -40,7 +40,7 @@
     context 'struct' do
       it 'uses struct constructor for member values' do
         module Test
-          class User < Dry::Types::Struct
+          class User < Dry::Struct
             attribute :name, 'string'
           end
         end
```
Comment (solnic) [inline spec/dry/types/array_spec.rb]: This can go

File: spec/dry/types/hash_spec.rb [hunk 5]
```diff
@@ -52,7 +52,7 @@ def self.name
       end
 
       module Test
-        class Location < Dry::Types::Value
+        class Location < Dry::Struct::Value
           attributes(lat: 'float', lng: 'float')
         end
       end
```
Comment (solnic) [inline spec/dry/types/hash_spec.rb]: and this can go too :)

File: Gemfile (hunk 2 above)
Comment (backus) [inline Gemfile]: We will have to release a dry-struct gem then

**Comment by backus** [comment]
@solnic should be ready for another pass

**Comment by backus** [comment]
@solnic ping

**Comment by solnic** [comment]
@backus thanks, could you update CHANGELOG in master?

**Comment by backus** [comment]
@solnic done: https://github.com/dry-rb/dry-types/commit/18f8928d70c3583f601f044cea4e48f8e7518e09

**Pull request merged**